### PR TITLE
Fix routing rule query rewrite

### PR DESCRIPTION
**Pull request by harshit-gangal**
## Description

This PR fixes the issue with query rewrite to also remove the table name qualifier.

## Related Issue(s)

- Fixes https://github.com/vitessio/vitess/issues/15248

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [x] If this change is to be back-ported to previous releases, a justification is included in the PR description
[ Regression in v18 and v19]
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on CI?
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/vt/vtgate/planbuilder/testdata/onecase.json [hunk 1]
```diff
@@ -1,9 +1,75 @@
...
+        "Query": "insert into music(id, user_id) values (:_id_0, :_user_id_0)",
+        "TableName": "music",
+        "VindexValues": {
+          "music_user_map": "2",
+          "user_index": "null"
+        }
+      },
+      "TablesUsed": [
+        "user.music"
+      ]
```
Comment (frouioui) [inline go/vt/vtgate/planbuilder/testdata/onecase.json:72]:
```suggestion
```

File: go/vt/vtgate/planbuilder/testdata/onecase.json [hunk 2]
```diff
@@ -1,9 +1,75 @@
 [
   {
     "comment": "Add your test case here for debugging and run go test -run=One.",
-    "query": "",
+    "query": "delete from second_user.bar",
```
Comment (frouioui) [inline go/vt/vtgate/planbuilder/testdata/onecase.json:4]:
```suggestion
    "query": "",
```

File: go/vt/vtgate/planbuilder/testdata/onecase.json [hunk 3]
```diff
@@ -3,7 +3,6 @@
     "comment": "Add your test case here for debugging and run go test -run=One.",
     "query": "",
     "plan": {
-
     }
```
Comment (frouioui) [inline go/vt/vtgate/planbuilder/testdata/onecase.json:6]:
```suggestion

    }
```

**Review by frouioui — APPROVED** [review]

File: go/vt/vtgate/planbuilder/testdata/dml_cases.json [hunk 4]
```diff
@@ -5828,5 +5828,78 @@
     "comment": "update with derived table",
     "query": "update (select id from user) as u set id = 4",
     "plan": "VT03031: the target table (select id from `user`) as u of the UPDATE is not updatable"
+  },
+  {
+    "comment": "Add your test case here for debugging and run go test -run=One.",
```
Comment (GuptaManan100) [inline go/vt/vtgate/planbuilder/testdata/dml_cases.json:5833]: The comment needs to change

File: go/vt/vtgate/planbuilder/testdata/dml_cases.json [hunk 5]
```diff
@@ -5828,5 +5828,78 @@
...
+        "Query": "delete from music as bar",
+        "Table": "music"
+      },
+      "TablesUsed": [
+        "user.music"
+      ]
+    }
+  },
+  {
+    "comment": "Add your test case here for debugging and run go test -run=One.",
```
Comment (GuptaManan100) [inline go/vt/vtgate/planbuilder/testdata/dml_cases.json:5858]: samesies

File: go/vt/vtgate/planbuilder/testdata/dml_cases.json [hunk 6]
```diff
@@ -5828,5 +5828,78 @@
...
+        "Query": "update music as bar set col = 23",
+        "Table": "music"
+      },
+      "TablesUsed": [
+        "user.music"
+      ]
+    }
+  },
+  {
+    "comment": "Add your test case here for debugging and run go test -run=One.",
```
Comment (GuptaManan100) [inline go/vt/vtgate/planbuilder/testdata/dml_cases.json:5880]: and this too

**Review by GuptaManan100 — APPROVED** [review]
Rest LGTM

**Pull request merged**